mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraIndexTableHandler.java

```diff
@@ -105,6 +158,19 @@ public CassandraIndexTableHandler(CassandraMailboxRecentsDAO mailboxRecentDAO,
         return mailboxCounterDAO.decrementUnseenAndCount(mailboxId);
     }
 
+    private Mono<Void> decrementCountersOnDelete(CassandraId mailboxId, Collection<MessageMetaData> metaData) {
+        long seenCount = metaData.stream()
+            .map(MessageMetaData::getFlags)
+            .filter(flags -> flags.contains(Flags.Flag.SEEN))
+            .count();
+
+        return mailboxCounterDAO.remove(MailboxCounters.builder()
+            .mailboxId(mailboxId)
+            .count(metaData.size())
+            .unseen(seenCount)
```

Comment:
Putting the number of messages being seen as an unseen count?

Reply:
The message that were seen and are now unseen. That's expected, no bug here.